templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -488,7 +488,7 @@
      });
 
      function showHideCancelInfo(obj) {
-       if (obj.find(":selected").text() == 'Refunded' || obj.find(":selected").text() == 'Cancelled') {
+       if (obj.find(":selected").val() == '7' || obj.find(":selected").val() == '3') {
```

Comment:
Simpler is always better.

But while this is undoubtedly an improvement, it worries me a little that we are relying on these numbers "7" and "3". Maybe they'll never change in the system... but if they did it would break here.